composer/core/state.py

```diff
@@ -56,9 +57,29 @@ def _ensure_backwards_compatible_checkpointing(state_dict: Dict[str, Any]):
     'callbacks',
     'scaler',
     'timestamp',
+    'dataloader_state',
 ]
 
 
+class DataloaderState:
```

Comment:
I see what you're trying to do here (to make the dataloader compatible with the logic to handle classes that have a `.state_dict` method), but I think this wrapping makes it slightly harder to follow what's going on. 

Instead, would it be possible to check if `attribute_name == 'dataloader'` below and then call these functions directly?